### ISSUE TITLE
How to analyse sentiment according to the aspect term using infer_example.py?

### ISSUE THREAD

**Issue by andrikristian** [songyouwei/ABSA-PyTorch#25]
HI,

In the file infer_example.py, I have the following code

`def evaluate(self, raw_texts):\n
         ......\n
         aspect_seqs = [self.tokenizer.text_to_sequence('battery')] * len(raw_texts)\n
         ......`

`t_probs = inf.evaluate(['laptop is good but battery is bad'])\n
print(t_probs.argmax(axis=-1) - 1)`

why is the sentiment score = 1?

I tested the sentence 'laptop is good' and 'battery is bad and the output are 1 and -1 respectively. But when I combine the sentence together, the output is always 1 no matter the aspect term. 

The model I am using is AOA, and it is trained on the laptop reviews and the accuracy is 0.7304.

**Comment by GeneZC** [comment]
Maybe you should give the way you combined them so that we could further give it a guess.

**Comment by andrikristian** [comment]
`import torch
import torch.nn.functional as F
import argparse

from data_utils import build_tokenizer, build_embedding_matrix
from models import IAN, MemNet, ATAE_LSTM, AOA, TD_LSTM, TNet_LF


class Inferer:
    """A simple inference example"""
    def __init__(self, opt):
        self.opt = opt
        self.tokenizer = build_tokenizer(
            fnames=[opt.dataset_file['train'], opt.dataset_file['test']],
            max_seq_len=opt.max_seq_len,
            dat_fname='{0}_tokenizer.dat'.format(opt.dataset))
        embedding_matrix = build_embedding_matrix(
            word2idx=self.tokenizer.word2idx,
            embed_dim=opt.embed_dim,
            dat_fname='{0}_{1}_embedding_matrix.dat'.format(str(opt.embed_dim), opt.dataset))
        self.model = opt.model_class(embedding_matrix, opt)
        print('loading model {0} ...'.format(opt.model_name))
        self.model.load_state_dict(torch.load(opt.state_dict_path))
        self.model = self.model.to(opt.device)
        # switch model to evaluation mode
        self.model.eval()
        torch.autograd.set_grad_enabled(False)

    def evaluate(self, raw_texts):
        context_seqs = [self.tokenizer.text_to_sequence(raw_text.lower().strip()) for raw_text in raw_texts]
        aspect_seqs = [self.tokenizer.text_to_sequence('battery')] * len(raw_texts)
        context_indices = torch.tensor(context_seqs, dtype=torch.int64).to(self.opt.device)
        aspect_indices = torch.tensor(aspect_seqs, dtype=torch.int64).to(self.opt.device)

        t_inputs = [context_indices, aspect_indices]
        t_outputs = self.model(t_inputs)

        t_probs = F.softmax(t_outputs, dim=-1).cpu().numpy()
        return t_probs


if __name__ == '__main__':
    model_classes = {
        'atae_lstm': ATAE_LSTM,
        'ian': IAN,
        'memnet': MemNet,
        'aoa': AOA,
        'td_lstm': TD_LSTM,
        'tnet_lf': TNet_LF
    }
    # set your trained models here

    model_state_dict_paths = {
        'atae_lstm': 'state_dict/Atae_lstm_laptop_state_dict_training_5repeats/atae_lstm_laptop_acc0.6865_repeats0',
        'ian': 'state_dict/ian_laptop_state_dict_training_5repeats/ian_laptop_acc0.71_repeats0',
        'memnet': 'state_dict/memnet_laptop_state_dict_training_5repeats/memnet_laptop_acc0.71_repeats2',
        'aoa': 'state_dict/AOA_laptop_state_dict_training_5repeats/aoa_laptop_acc0.7304_repeats3',
        'td_lstm': 'state_dict/td_lstm_laptop_state_dict_training_5repeats/td_lstm_laptop_acc0.6975_repeats2',
        'tnet_lf': 'state_dict/tnet_lf_laptop_state_dict_training_5repeats/tnet_lf_laptop_acc0.7398_repeats3'
    }

    class Option(object): pass
    opt = Option()
    opt.model_name = 'aoa'
    opt.model_class = model_classes[opt.model_name]
    opt.dataset = 'laptop'
    opt.dataset_file = {
        'train': './datasets/semeval14/Laptops_Train.xml.seg',
        'test': './datasets/semeval14/Laptops_Test_Gold.xml.seg'
    }
    opt.state_dict_path = model_state_dict_paths[opt.model_name]
    opt.embed_dim = 300
    opt.hidden_dim = 300
    opt.max_seq_len = 80
    opt.polarities_dim = 3
    opt.hops = 3
    opt.device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

    inf = Inferer(opt)
    t_probs = inf.evaluate(['laptop is good but battery is bad'])
    print(t_probs)
    print(t_probs.argmax(axis=-1) - 1)
`

**Comment by GeneZC** [comment]
Generally, it's because both 'good' and 'bad' could be descriptive of 'battery', and most models only concentrate on semantic correlations, leading to the result of misidentifying the correct modifier.

**Comment by andrikristian** [comment]
Thanks. I found out that if I input 'good laptop, bad battery', the sentiment predictions are correct for both aspect terms: 'laptop' and 'battery'. 

Is it true that the context words behind the aspect word weigh more for sentiment prediction?

**Comment by GeneZC** [comment]
Actually, prior works mainly focus on how to obtain a aspect-specific context representation properly so that we could easily judge which part in the context is much more important than the rest.
In this way, from my perspective, aspect and the corresponding context are of equal importance.

**Comment by andrikristian** [comment]
ok thanks for your answers.
